spec/index.bs

```diff
@@ -1122,66 +1125,81 @@ This specifies the options to use in creating a {{GPUBuffer}}.
-
-                1. Record a validation error in the current scope.
-                <!-- TODO(kangz): Once we have a description of the error monad, explain what the error buffer is. -->
-                1. Create an invalid {{GPUBuffer}} and return the result.
-
-            1. Let |b| be a new {{GPUBuffer}} object.
-            1. Set the {{GPUBuffer/[[size]]}} slot of |b| to the value of the {{GPUBufferDescriptor/size}} attribute of |descriptor|.
-            1. Let |m| be a zero-filled {{ArrayBuffer}} of size the {{GPUBuffer/[[size]]}} slot of |b|.
-            1. Set the {{GPUBuffer/[[usage]]}} slot of |b| to the value of the {{GPUBufferDescriptor/usage}} attribute of |descriptor|.
-            1. Set the {{GPUBuffer/[[state]]}} internal slot of |b| to [=buffer state/mapped for writing=].
-            1. Set the {{GPUBuffer/[[mapping]]}} internal slot of |b| to |m|.
-            1. Set each byte of |b|'s allocation to zero.
-            1. Return a sequence containing |b| and |m| in that order.
-        </div>
-</dl>
+  1. If this call doesn't follow the [$createBuffer Valid Usage$]:
+
+    Issue(gpuweb/gpuweb#605): Explain that the resulting error buffer can still be mapped at creation.
+
+  1. Let |b| be a new {{GPUBuffer}} object.
+  1. Set |b|.{{GPUBuffer/[[size]]}} to |descriptor|.{{GPUBufferDescriptor/size}}.
+  1. Set |b|.{{GPUBuffer/[[usage]]}} to |descriptor|.{{GPUBufferDescriptor/usage}}.
+  1. If |descriptor|.{{GPUBufferDescriptor/mappedAtCreation}} if true:
+
+    1. Set |b|.{{GPUBuffer/[[mapping]]}} to a new {{ArrayBuffer}} of size |b|.{{GPUBuffer/[[size]]}}.
+    1. Set |b|.{{GPUBuffer/[[mapping_range]]}} to `[0, descriptor.size]`.
+    1. Set |b|.{{GPUBuffer/[[mapped_ranges]]}} to `[]`.
+    1. Set |b|.{{GPUBuffer/[[state]]}} to [=buffer state/mapped at creation=].
+
+  1. Else:
+
+    1. Set |b|.{{GPUBuffer/[[mapping]]}} to `null`.
+    1. Set |b|.{{GPUBuffer/[[mapping_range]]}} to `null`.
+    1. Set |b|.{{GPUBuffer/[[mapped_ranges]]}} to `null`.
+    1. Set |b|.{{GPUBuffer/[[state]]}} to [=buffer state/unmapped=].
+
+  1. Set each byte of |b|'s allocation to zero.
+  1. Return |b|.
+
+  Note: it is valid to set {{GPUBufferDescriptor/mappedAtCreation}} to true without {{GPUBufferUsage/MAP_READ}}
+    or {{GPUBufferUsage/MAP_WRITE}} in {{GPUBufferDescriptor/usage}}. This can be used to set the buffer's
+    initial data.
+
+  <div algorithm class=validusage>
+    <dfn abstract-op>createBuffer Valid Usage</dfn>
+      Given a {{GPUDevice}} |this| and a {{GPUBufferDescriptor}} |descriptor|
+      the following validation rules apply:
+
+      1, |this| must be a [=valid=] {{GPUDevice}}.
+      1. |descriptor|.{{GPUBufferDescriptor/usage}} must be a subset of |this|.[[allowed buffer usages]].
+      1. If |descriptor|.{{GPUBufferDescriptor/usage}} contains {{GPUBufferUsage/MAP_READ}} then
+        it must be a subset of {{GPUBufferUsage/MAP_READ}} | {{GPUBufferUsage/COPY_DST}}.
```

Comment:
While I agree, that's not the takeaway from last call. If we need to debate this more, let's do it independently of this PR.

Reply:
I don't think there was a strong yes from you or anybody on the need to have a single call. It was rather "maybe?", and now that we see cleaner how this would look (thanks for the PR!) at least on my side the "maybe?" leans more towards "no".

Reply:
We discussed this offline with @kvark @kainino0x and @jdashg another possibility would be to add a `MapFlags` argument to `mapAsync` that would have to match the `MAP_*` usage for now. It allows future extensibility and makes it clear at the map call which it is.